src/snowflake/connector/storage_client.py

```diff
@@ -358,9 +358,18 @@ def prepare_download(self) -> None:
     def write_downloaded_chunk(self, chunk_id: int, data: bytes) -> None:
         """Writes given data to the temp location starting at chunk_id * chunk_size."""
         # TODO: should we use chunking and write content in smaller chunks?
-        with self.intermediate_dst_path.open("rb+") as fd:
-            fd.seek(self.chunk_size * chunk_id)
-            fd.write(data)
+        try:
+            with self.intermediate_dst_path.open("rb+") as fd:
+                fd.seek(self.chunk_size * chunk_id)
+                fd.write(data)
+        except FileNotFoundError:
+            # we don't maintain dir structure when downloading, making it possible that we download
+            # same file twice, which cause race condition and operationalError in sync client because of multi-process
+            # design in sync client.
+            # while in async client, everything is actually sync, making same name file being deleted and
+            # cause file not found error
+            # TODO: this is to align with sync file transfer
+            raise OperationalError
```

Comment:
why are we changing the sync one here.. aren't we supposed to update the async one?

Reply:
we need to have a separate one for aio